CMake 3.16.0: Mandatory test fails

After upgrading `CMake` from version `3.15.5` to `3.16.0`, the mandatory test `DEAL_II_HAVE_USABLE_FLAGS_DEBUG` fails on my machine with the following `CMakeError.log`. Configured with `MPI=ON, P4EST=ON, TRILINOS=ON` together with `gcc 9.2.0` and `openmpi 4.0.2`.

---

```
Performing C++ SOURCE FILE Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG failed with the following output:
Change Dir: /home/marc/work/dealii/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_db18c/fast && /usr/bin/make -f CMakeFiles/cmTC_db18c.dir/build.make CMakeFiles/cmTC_db18c.dir/build
make[1]: Entering directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_db18c.dir/src.cxx.o
/usr/bin/c++    -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -pthread -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections   -o CMakeFiles/cmTC_db18c.dir/src.cxx.o -c /home/marc/work/dealii/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTC_db18c
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_db18c.dir/link.txt --verbose=1
/usr/bin/c++   -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -pthread -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections    -rdynamic CMakeFiles/cmTC_db18c.dir/src.cxx.o  -o cmTC_db18c  -Wl,-rpath,/usr/lib/openmpi:/home/marc/bin/trilinos-12.14.1/lib:/home/marc/bin/p4est-2.2/DEBUG/lib -rdynamic -fuse-ld=gold -Wl,-rpath -Wl,/usr/lib/openmpi -Wl,--enable-new-dtags -pthread pthread -ggdb /usr/lib/openmpi/libmpi_usempif08.so /usr/lib/openmpi/libmpi_usempi_ignore_tkr.so /usr/lib/openmpi/libmpi_mpifh.so /usr/lib/libz.so /usr/lib/libboost_iostreams.so /usr/lib/libboost_serialization.so /usr/lib/libboost_system.so /usr/lib/libboost_thread.so /usr/lib/libboost_regex.so /usr/lib/libboost_chrono.so /usr/lib/libboost_date_time.so /usr/lib/libboost_atomic.so /home/marc/bin/trilinos-12.14.1/lib/libmuelu-adapters.so /home/marc/bin/trilinos-12.14.1/lib/libmuelu-interface.so /home/marc/bin/trilinos-12.14.1/lib/libmuelu.so /home/marc/bin/trilinos-12.14.1/lib/libintrepid2.so /home/marc/bin/trilinos-12.14.1/lib/libteko.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikos.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosbelos.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosamesos2.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosaztecoo.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosamesos.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosml.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosifpack.so /home/marc/bin/trilinos-12.14.1/lib/libifpack2-adapters.so /home/marc/bin/trilinos-12.14.1/lib/libifpack2.so /home/marc/bin/trilinos-12.14.1/lib/libanasazitpetra.so /home/marc/bin/trilinos-12.14.1/lib/libModeLaplace.so /home/marc/bin/trilinos-12.14.1/lib/libanasaziepetra.so /home/marc/bin/trilinos-12.14.1/lib/libanasazi.so /home/marc/bin/trilinos-12.14.1/lib/libamesos2.so /home/marc/bin/trilinos-12.14.1/lib/libshylu_nodetacho.so /home/marc/bin/trilinos-12.14.1/lib/libshylu_nodehts.so /home/marc/bin/trilinos-12.14.1/lib/libbelosxpetra.so /home/marc/bin/trilinos-12.14.1/lib/libbelostpetra.so /home/marc/bin/trilinos-12.14.1/lib/libbelosepetra.so /home/marc/bin/trilinos-12.14.1/lib/libbelos.so /home/marc/bin/trilinos-12.14.1/lib/libml.so /home/marc/bin/trilinos-12.14.1/lib/libifpack.so /home/marc/bin/trilinos-12.14.1/lib/libzoltan2.so /home/marc/bin/trilinos-12.14.1/lib/libpamgen_extras.so /home/marc/bin/trilinos-12.14.1/lib/libpamgen.so /home/marc/bin/trilinos-12.14.1/lib/libamesos.so /home/marc/bin/trilinos-12.14.1/lib/libgaleri-xpetra.so /home/marc/bin/trilinos-12.14.1/lib/libgaleri-epetra.so /home/marc/bin/trilinos-12.14.1/lib/libaztecoo.so /home/marc/bin/trilinos-12.14.1/lib/libisorropia.so /home/marc/bin/trilinos-12.14.1/lib/libxpetra-sup.so /home/marc/bin/trilinos-12.14.1/lib/libxpetra.so /home/marc/bin/trilinos-12.14.1/lib/libthyratpetra.so /home/marc/bin/trilinos-12.14.1/lib/libthyraepetraext.so /home/marc/bin/trilinos-12.14.1/lib/libthyraepetra.so /home/marc/bin/trilinos-12.14.1/lib/libthyracore.so /home/marc/bin/trilinos-12.14.1/lib/libepetraext.so /home/marc/bin/trilinos-12.14.1/lib/libtrilinosss.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraext.so /home/marc/bin/trilinos-12.14.1/lib/libtpetrainout.so /home/marc/bin/trilinos-12.14.1/lib/libtpetra.so /home/marc/bin/trilinos-12.14.1/lib/libkokkostsqr.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraclassiclinalg.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraclassicnodeapi.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraclassic.so /home/marc/bin/trilinos-12.14.1/lib/libtriutils.so /home/marc/bin/trilinos-12.14.1/lib/libshards.so /home/marc/bin/trilinos-12.14.1/lib/libzoltan.so /home/marc/bin/trilinos-12.14.1/lib/libepetra.so /home/marc/bin/trilinos-12.14.1/lib/libsacado.so /home/marc/bin/trilinos-12.14.1/lib/librtop.so /home/marc/bin/trilinos-12.14.1/lib/libkokkoskernels.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoskokkoscomm.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoskokkoscompat.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosremainder.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosnumerics.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoscomm.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosparameterlist.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosparser.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoscore.so /home/marc/bin/trilinos-12.14.1/lib/libkokkosalgorithms.so /home/marc/bin/trilinos-12.14.1/lib/libkokkoscontainers.so /home/marc/bin/trilinos-12.14.1/lib/libkokkoscore.so /home/marc/bin/trilinos-12.14.1/lib/libgtest.so /usr/lib/libblas.so -ldl /usr/lib/openmpi/libmpi_cxx.so /usr/lib/libumfpack.so /usr/lib/libcholmod.so /usr/lib/libccolamd.so /usr/lib/libcolamd.so /usr/lib/libcamd.so /usr/lib/libsuitesparseconfig.so /usr/lib/libamd.so /usr/lib/libmetis.so -lrt /usr/lib/libgsl.so /usr/lib/libgslcblas.so /usr/lib/libhdf5_hl.so /usr/lib/libhdf5.so /usr/lib/liblapack.so /usr/lib/libopenblas.so -lgfortran -lquadmath -lm -lc -lgcc_s -lgcc /usr/lib/openmpi/libmpi.so -ltbb_debug /home/marc/bin/p4est-2.2/DEBUG/lib/libp4est.so /home/marc/bin/p4est-2.2/DEBUG/lib/libsc.so 
c++: error: pthread: No such file or directory
make[1]: *** [CMakeFiles/cmTC_db18c.dir/build.make:188: cmTC_db18c] Error 1
make[1]: Leaving directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:121: cmTC_db18c/fast] Error 2


Source file was:
int main(){ return 0; }
Performing C++ SOURCE FILE Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG failed with the following output:
Change Dir: /home/marc/work/dealii/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_0eb22/fast && /usr/bin/make -f CMakeFiles/cmTC_0eb22.dir/build.make CMakeFiles/cmTC_0eb22.dir/build
make[1]: Entering directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_0eb22.dir/src.cxx.o
/usr/bin/c++    -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -pthread -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections   -o CMakeFiles/cmTC_0eb22.dir/src.cxx.o -c /home/marc/work/dealii/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTC_0eb22
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_0eb22.dir/link.txt --verbose=1
/usr/bin/c++   -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -pthread -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections    -rdynamic CMakeFiles/cmTC_0eb22.dir/src.cxx.o  -o cmTC_0eb22  -Wl,-rpath,/usr/lib/openmpi:/home/marc/bin/trilinos-12.14.1/lib:/home/marc/bin/p4est-2.2/DEBUG/lib -rdynamic  -Wl,-rpath -Wl,/usr/lib/openmpi -Wl,--enable-new-dtags -pthread pthread -ggdb /usr/lib/openmpi/libmpi_usempif08.so /usr/lib/openmpi/libmpi_usempi_ignore_tkr.so /usr/lib/openmpi/libmpi_mpifh.so /usr/lib/libz.so /usr/lib/libboost_iostreams.so /usr/lib/libboost_serialization.so /usr/lib/libboost_system.so /usr/lib/libboost_thread.so /usr/lib/libboost_regex.so /usr/lib/libboost_chrono.so /usr/lib/libboost_date_time.so /usr/lib/libboost_atomic.so /home/marc/bin/trilinos-12.14.1/lib/libmuelu-adapters.so /home/marc/bin/trilinos-12.14.1/lib/libmuelu-interface.so /home/marc/bin/trilinos-12.14.1/lib/libmuelu.so /home/marc/bin/trilinos-12.14.1/lib/libintrepid2.so /home/marc/bin/trilinos-12.14.1/lib/libteko.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikos.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosbelos.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosamesos2.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosaztecoo.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosamesos.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosml.so /home/marc/bin/trilinos-12.14.1/lib/libstratimikosifpack.so /home/marc/bin/trilinos-12.14.1/lib/libifpack2-adapters.so /home/marc/bin/trilinos-12.14.1/lib/libifpack2.so /home/marc/bin/trilinos-12.14.1/lib/libanasazitpetra.so /home/marc/bin/trilinos-12.14.1/lib/libModeLaplace.so /home/marc/bin/trilinos-12.14.1/lib/libanasaziepetra.so /home/marc/bin/trilinos-12.14.1/lib/libanasazi.so /home/marc/bin/trilinos-12.14.1/lib/libamesos2.so /home/marc/bin/trilinos-12.14.1/lib/libshylu_nodetacho.so /home/marc/bin/trilinos-12.14.1/lib/libshylu_nodehts.so /home/marc/bin/trilinos-12.14.1/lib/libbelosxpetra.so /home/marc/bin/trilinos-12.14.1/lib/libbelostpetra.so /home/marc/bin/trilinos-12.14.1/lib/libbelosepetra.so /home/marc/bin/trilinos-12.14.1/lib/libbelos.so /home/marc/bin/trilinos-12.14.1/lib/libml.so /home/marc/bin/trilinos-12.14.1/lib/libifpack.so /home/marc/bin/trilinos-12.14.1/lib/libzoltan2.so /home/marc/bin/trilinos-12.14.1/lib/libpamgen_extras.so /home/marc/bin/trilinos-12.14.1/lib/libpamgen.so /home/marc/bin/trilinos-12.14.1/lib/libamesos.so /home/marc/bin/trilinos-12.14.1/lib/libgaleri-xpetra.so /home/marc/bin/trilinos-12.14.1/lib/libgaleri-epetra.so /home/marc/bin/trilinos-12.14.1/lib/libaztecoo.so /home/marc/bin/trilinos-12.14.1/lib/libisorropia.so /home/marc/bin/trilinos-12.14.1/lib/libxpetra-sup.so /home/marc/bin/trilinos-12.14.1/lib/libxpetra.so /home/marc/bin/trilinos-12.14.1/lib/libthyratpetra.so /home/marc/bin/trilinos-12.14.1/lib/libthyraepetraext.so /home/marc/bin/trilinos-12.14.1/lib/libthyraepetra.so /home/marc/bin/trilinos-12.14.1/lib/libthyracore.so /home/marc/bin/trilinos-12.14.1/lib/libepetraext.so /home/marc/bin/trilinos-12.14.1/lib/libtrilinosss.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraext.so /home/marc/bin/trilinos-12.14.1/lib/libtpetrainout.so /home/marc/bin/trilinos-12.14.1/lib/libtpetra.so /home/marc/bin/trilinos-12.14.1/lib/libkokkostsqr.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraclassiclinalg.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraclassicnodeapi.so /home/marc/bin/trilinos-12.14.1/lib/libtpetraclassic.so /home/marc/bin/trilinos-12.14.1/lib/libtriutils.so /home/marc/bin/trilinos-12.14.1/lib/libshards.so /home/marc/bin/trilinos-12.14.1/lib/libzoltan.so /home/marc/bin/trilinos-12.14.1/lib/libepetra.so /home/marc/bin/trilinos-12.14.1/lib/libsacado.so /home/marc/bin/trilinos-12.14.1/lib/librtop.so /home/marc/bin/trilinos-12.14.1/lib/libkokkoskernels.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoskokkoscomm.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoskokkoscompat.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosremainder.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosnumerics.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoscomm.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosparameterlist.so /home/marc/bin/trilinos-12.14.1/lib/libteuchosparser.so /home/marc/bin/trilinos-12.14.1/lib/libteuchoscore.so /home/marc/bin/trilinos-12.14.1/lib/libkokkosalgorithms.so /home/marc/bin/trilinos-12.14.1/lib/libkokkoscontainers.so /home/marc/bin/trilinos-12.14.1/lib/libkokkoscore.so /home/marc/bin/trilinos-12.14.1/lib/libgtest.so /usr/lib/libblas.so -ldl /usr/lib/openmpi/libmpi_cxx.so /usr/lib/libumfpack.so /usr/lib/libcholmod.so /usr/lib/libccolamd.so /usr/lib/libcolamd.so /usr/lib/libcamd.so /usr/lib/libsuitesparseconfig.so /usr/lib/libamd.so /usr/lib/libmetis.so -lrt /usr/lib/libgsl.so /usr/lib/libgslcblas.so /usr/lib/libhdf5_hl.so /usr/lib/libhdf5.so /usr/lib/liblapack.so /usr/lib/libopenblas.so -lgfortran -lquadmath -lm -lc -lgcc_s -lgcc /usr/lib/openmpi/libmpi.so -ltbb_debug /home/marc/bin/p4est-2.2/DEBUG/lib/libp4est.so /home/marc/bin/p4est-2.2/DEBUG/lib/libsc.so 
c++: error: pthread: No such file or directory
make[1]: *** [CMakeFiles/cmTC_0eb22.dir/build.make:188: cmTC_0eb22] Error 1
make[1]: Leaving directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:121: cmTC_0eb22/fast] Error 2


Source file was:
int main(){ return 0; }
```

---

It appears that the most recent `CMake` does not get a reference to the `POSIX Threads` API.

Test `DEAL_II_HAVE_FLAG_pthread` passes, while `DEAL_II_HAVE_USABLE_FLAGS_DEBUG` does not.

I reduced my cmake configuration to the default one, which sets `DEAL_II_WITH_THREADS=ON`. Let me try to be more verbose on the configuration results:

---

`CMake` console output
```
[...]
-- Performing Test DEAL_II_HAVE_FLAG_pthread
-- Performing Test DEAL_II_HAVE_FLAG_pthread - Success
[...]
-- Include /home/marc/work/dealii/cmake/setup_finalize.cmake
-- Performing Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG
-- Performing Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG - Failed
-- Unable to compile a simple test program. Trying to drop "-fuse-ld=gold" from the linker flags.
-- Performing Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG
-- Performing Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG - Failed
CMake Error at cmake/setup_finalize.cmake:137 (MESSAGE):
  

    Configuration error: Cannot compile a test program with the final set of
    compiler and linker flags:
      CXX flags (DEBUG): -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections
      LD flags  (DEBUG): -Wl,--as-needed -rdynamic  pthread -ggdb
      LIBRARIES (DEBUG): /usr/lib/libz.so;/usr/lib/libboost_iostreams.so;/usr/lib/libboost_serialization.so;/usr/lib/libboost_system.so;/usr/lib/libboost_thread.so;/usr/lib/libboost_regex.so;/usr/lib/libboost_chrono.so;/usr/lib/libboost_date_time.so;/usr/lib/libboost_atomic.so;/usr/lib/libumfpack.so;/usr/lib/libcholmod.so;/usr/lib/libccolamd.so;/usr/lib/libcolamd.so;/usr/lib/libcamd.so;/usr/lib/libsuitesparseconfig.so;/usr/lib/libamd.so;/usr/lib/libmetis.so;/usr/lib/liblapack.so;/usr/lib/libopenblas.so;gfortran;quadmath;m;c;gcc_s;gcc;rt;/usr/lib/libgsl.so;/usr/lib/libgslcblas.so;/usr/lib/libtbb_debug.so
    

  

Call Stack (most recent call first):
  cmake/macros/macro_verbose_include.cmake:19 (INCLUDE)
  CMakeLists.txt:132 (VERBOSE_INCLUDE)


-- Configuring incomplete, errors occurred!
See also "/home/marc/work/dealii/build/CMakeFiles/CMakeOutput.log".
See also "/home/marc/work/dealii/build/CMakeFiles/CMakeError.log".
```

---

`CMakeFiles/CMakeOutput.log`
```
[...]
Performing C++ SOURCE FILE Test DEAL_II_HAVE_FLAG_pthread succeeded with the following output:
Change Dir: /home/marc/work/dealii/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_6464e/fast && /usr/bin/make -f CMakeFiles/cmTC_6464e.dir/build.make CMakeFiles/cmTC_6464e.dir/build
make[1]: Entering directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_6464e.dir/src.cxx.o
/usr/bin/c++    -DDEAL_II_HAVE_FLAG_pthread   -pthread -o CMakeFiles/cmTC_6464e.dir/src.cxx.o -c /home/marc/work/dealii/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTC_6464e
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_6464e.dir/link.txt --verbose=1
/usr/bin/c++   -DDEAL_II_HAVE_FLAG_pthread    -rdynamic CMakeFiles/cmTC_6464e.dir/src.cxx.o  -o cmTC_6464e  /usr/lib/liblapack.so /usr/lib/libopenblas.so -lgfortran -lquadmath -lm -lc -lgcc_s -lgcc 
make[1]: Leaving directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'


Source file was:
int main() { return 0; }
[...]
```

---

`CMakeFiles/CMakeError.log`
```
[...]
Performing C++ SOURCE FILE Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG failed with the following output:
Change Dir: /home/marc/work/dealii/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_cfe1d/fast && /usr/bin/make -f CMakeFiles/cmTC_cfe1d.dir/build.make CMakeFiles/cmTC_cfe1d.dir/build
make[1]: Entering directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_cfe1d.dir/src.cxx.o
/usr/bin/c++    -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections   -o CMakeFiles/cmTC_cfe1d.dir/src.cxx.o -c /home/marc/work/dealii/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTC_cfe1d
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_cfe1d.dir/link.txt --verbose=1
/usr/bin/c++   -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections    -rdynamic CMakeFiles/cmTC_cfe1d.dir/src.cxx.o  -o cmTC_cfe1d  -rdynamic -fuse-ld=gold pthread -ggdb /usr/lib/libz.so /usr/lib/libboost_iostreams.so /usr/lib/libboost_serialization.so /usr/lib/libboost_system.so /usr/lib/libboost_thread.so /usr/lib/libboost_regex.so /usr/lib/libboost_chrono.so /usr/lib/libboost_date_time.so /usr/lib/libboost_atomic.so /usr/lib/libumfpack.so /usr/lib/libcholmod.so /usr/lib/libccolamd.so /usr/lib/libcolamd.so /usr/lib/libcamd.so /usr/lib/libsuitesparseconfig.so /usr/lib/libamd.so /usr/lib/libmetis.so /usr/lib/liblapack.so /usr/lib/libopenblas.so -lgfortran -lquadmath -lm -lc -lgcc_s -lgcc -lrt /usr/lib/libgsl.so /usr/lib/libgslcblas.so -ltbb_debug 
c++: error: pthread: No such file or directory
make[1]: *** [CMakeFiles/cmTC_cfe1d.dir/build.make:109: cmTC_cfe1d] Error 1
make[1]: Leaving directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:121: cmTC_cfe1d/fast] Error 2


Source file was:
int main(){ return 0; }
Performing C++ SOURCE FILE Test DEAL_II_HAVE_USABLE_FLAGS_DEBUG failed with the following output:
Change Dir: /home/marc/work/dealii/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_9c66e/fast && /usr/bin/make -f CMakeFiles/cmTC_9c66e.dir/build.make CMakeFiles/cmTC_9c66e.dir/build
make[1]: Entering directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
Building CXX object CMakeFiles/cmTC_9c66e.dir/src.cxx.o
/usr/bin/c++    -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections   -o CMakeFiles/cmTC_9c66e.dir/src.cxx.o -c /home/marc/work/dealii/build/CMakeFiles/CMakeTmp/src.cxx
Linking CXX executable cmTC_9c66e
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_9c66e.dir/link.txt --verbose=1
/usr/bin/c++   -DDEAL_II_HAVE_USABLE_FLAGS_DEBUG -pedantic -fPIC -Wall -Wextra -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -std=c++17 -Wno-unused-local-typedefs -O0 -ggdb -Wa,--compress-debug-sections    -rdynamic CMakeFiles/cmTC_9c66e.dir/src.cxx.o  -o cmTC_9c66e  -rdynamic  pthread -ggdb /usr/lib/libz.so /usr/lib/libboost_iostreams.so /usr/lib/libboost_serialization.so /usr/lib/libboost_system.so /usr/lib/libboost_thread.so /usr/lib/libboost_regex.so /usr/lib/libboost_chrono.so /usr/lib/libboost_date_time.so /usr/lib/libboost_atomic.so /usr/lib/libumfpack.so /usr/lib/libcholmod.so /usr/lib/libccolamd.so /usr/lib/libcolamd.so /usr/lib/libcamd.so /usr/lib/libsuitesparseconfig.so /usr/lib/libamd.so /usr/lib/libmetis.so /usr/lib/liblapack.so /usr/lib/libopenblas.so -lgfortran -lquadmath -lm -lc -lgcc_s -lgcc -lrt /usr/lib/libgsl.so /usr/lib/libgslcblas.so -ltbb_debug 
c++: error: pthread: No such file or directory
make[1]: *** [CMakeFiles/cmTC_9c66e.dir/build.make:109: cmTC_9c66e] Error 1
make[1]: Leaving directory '/home/marc/work/dealii/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:121: cmTC_9c66e/fast] Error 2


Source file was:
int main(){ return 0; }
```

--- a/cmake/configure/configure_1_threads.cmake
+++ b/cmake/configure/configure_1_threads.cmake
@@ -38,6 +38,16 @@ MACRO(SETUP_THREADING)
 
     RESET_CMAKE_REQUIRED()
 
+    #
+    # The FindThreads macro returned a linker option instead of the actual
+    # library name in earlier versions. We still require the linker option,
+    # so we fix the corresponding variable.
+    #  - See: https://gitlab.kitware.com/cmake/cmake/issues/19747
+    #
+    IF(CMAKE_THREAD_LIBS_INIT AND NOT "${CMAKE_THREAD_LIBS_INIT}" MATCHES "^-l")
+      STRING(PREPEND CMAKE_THREAD_LIBS_INIT "-l")
+    ENDIF()
+
   ELSE()
 
     #

